src/material/dialog/dialog-container.ts

```diff
@@ -96,6 +101,7 @@ export class MatDialogContainer extends CdkDialogContainer<MatDialogConfig> impl
 
   constructor(
     elementRef: ElementRef,
+    _platform: Platform,
```

Comment:
This is an unnecessary breaking change. The `Platform` is injected in the `CdkDialogContainer`, you should mark it as `protected` instead of `private`. Or implement this a11y feature right in the `CdkDialogContainer`?